### ISSUE TITLE
Allow user to access the randomly-generated WinRM password via the template or environment.

### ISSUE THREAD

**Issue by SwampDragons** [hashicorp/packer#5895]
We recommend using shell-local to run Ansible, but doing so means users must first generate the temporary host file.  In https://github.com/hashicorp/packer/issues/4904#issuecomment-349844919 a user requested being able to access the randomly-generated EC2 Windows password in order to be able to generate this file. 

Feature:  Add a new template or env variable like "{{ .WinRMPassword }}"

**Comment by matt-richardson** [comment]
Will this be ansible specific? This would be super useful in the `powershell` provisioner as well.

**Comment by SwampDragons** [comment]
@matt-richardson good point; I'll remove the ansible tag from this issue.

**Comment by rickard-von-essen** [comment]
See #3911 and #3969

**Comment by SwampDragons** [comment]
https://github.com/hashicorp/packer/pull/4409 is probably the right approach here.

**Comment by Marcus-James-Adams** [comment]
I have a similar request under https://github.com/hashicorp/packer/issues/5871 for oracle-oci i also know that alibaba and some openstack also have the random password generated at build - perhaps this should be an option for all providers?

**Comment by SwampDragons** [comment]
@Marcus-James-Adams Agreed. Once we have the framework in place for AWS it shouldn't be too hard to enable it for other providers as well.

**Comment by liberforce** [comment]
This was at first open for the `ansible` provisioner, and it ended fixed for the `powershell` provisionner but not the `ansible`, nor `shell-local` ones. As stated in #5998 :

> that issue won't be truly addressed until you can also access this variable from the shell-local provisioner

Please reopen.

**Comment by SwampDragons** [comment]
Good point I've opened separate tickets for each place we implement this, just to keep scope clear. I'll change the title of this one to be specific to Powershell.

**Comment by SwampDragons** [comment]
@liberforce if you'd like to test the patch enabling you to access the winrm password from the shell-local provisioner, check out #6251.  I can build it for you if you let me know your host architecture.

**Comment by liberforce** [comment]
@SwampDragons : sorry for the late reply, I was out of office for the passed month. Thanks for your proposal, I'll be happy to test a binary if you can still provide it. I'm running an x86_64 Ubuntu 14.04.

**Comment by SwampDragons** [comment]
@liberforce you don't need a binary; this has been released and is in Packer 1.2.4
